src/signalr-hub.ts

```diff
@@ -56,8 +57,11 @@ export class SignalRHub {
         return subject.asObservable();
     }
 
-    send<T>(method: string, data: T) {
-        this.proxy.invoke(method, data);
+    async send(method: string, ...args: any[]): Promise<any> {
```

Comment:
What about using an Observable as the return here as well. To keep the public facing interface using observables.

Reply:
In this case I also chose a promise over an observable for a reason. You are right though that this somehow breaks the public facing interface by introducing a new return type.
Again I think however that it is useful to be able to await the send method, because that would make the code using the component much more lean and readable in my opinion.

But here's a proposal: Make the send method return an Observable so this does not break the interface and return a new method named `sendAsync` that returns a promise, so the method can be invoked using the await keyword.

Reply:
One could even have the `sendAsync`-method use the `send`-method and invoke the `ToPromise`-operator of rxjs.